src/Materials/Node/Blocks/Vertex/instancesBlock.ts

```diff
@@ -175,7 +176,11 @@ export class InstancesBlock extends NodeMaterialBlock {
         state.compilationString += `#ifdef THIN_INSTANCES\r\n`;
         state.compilationString += `${output.associatedVariableName} = ${this.world.associatedVariableName} * ${output.associatedVariableName};\r\n`;
         state.compilationString += `#endif\r\n`;
-        state.compilationString += this._declareOutput(instanceID, state) + ` = float(gl_InstanceID);\r\n`;
+        if (Engine.LastCreatedEngine && !Engine.LastCreatedEngine._caps.canUseGLInstanceID) {

```

Comment:
Can we instead access it from the scene ? if would prefer to not rely on the lastCreated one ?

Reply:
The problem is that I have no scene / engine instance available in the class... That's why I used the last created engine.

Reply:
Wondering if are any way to pass the information in ? do not worry if it is too complex ?

Reply:
Ok, I was able to find the scene somewhere :)